lib/src/main/kotlin/net/dean/jraw/RedditClient.kt

```diff
@@ -321,12 +325,33 @@ class RedditClient internal constructor(
     @EndpointImplementation(Endpoint.GET_RANDOM, type = MethodType.NON_BLOCKING_CALL)
     fun randomSubreddit() = subreddit("random")
 
+    /**
+     * Creates a PublicContributionReference (either SubmissionReference or CommentReference).
+     * The expected input is provided in format KIND_ID (e.g. t3_6afe8u or t1_2bad4u)
+     */
+    fun publicContribution(fullname: String): PublicContributionReference {
```

Comment:
Preface - my use case is a script app (i.e. reddit bot running 24/7). Mostly the use case has been to get a reference to comment's parent -
 `Comment.getParentFullName()` returns a fullname (`String`) and you have to manually detect the kind of the parent to then call either `submission()` or `comment()` even when you don't care which one it is (as PublicContributionReference provides a lot of actions one may need, e.g. `reply()`, `edit()`,  `delete()`, and `remove()`.

I have updated the documentation on that method ( https://github.com/mattbdean/JRAW/pull/207/commits/136eb1ca7e1883d52aba3781c01f8612c2aefb51 ), let me know if you think this is something the end-users should define themselves as a helper/extension function instead of having it in the library API.

Reply:
In that case this method makes sense. Our goal is to make this as easy as possible for the developers using the library.